lib/chef/resource/launchd.rb

```diff
@@ -0,0 +1,285 @@
+#
+# Author:: Mike Dodge (<mikedodge04@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource/file'
+require 'chef/provider/launchd'
+require 'chef/mixin/securable'
+
+class Chef
+  class Resource
+    class Launchd < Chef::Resource::File
+      include Chef::Mixin::Securable
+
+      provides :launchd
+
+      def initialize(name, run_context = nil)
+        super
+        @provider = Chef::Provider::Launchd
+        @resource_name = :launchd
+        @action = 'create'
+        @cookbook = nil
+        @source = nil
+        @label = name
+        @disabled = false
+        @service_type = 'system'
+        @path = "/Library/LaunchDaemons/#{name}.plist"
+      end
+
+      def label(label = @label)
+        set_or_return(:label, label, :kind_of => String)
+      end
+
+      def source(source_filename = nil)
+        set_or_return(:source, source_filename, :kind_of => [String, Array])
+      end
+
+      def cookbook(cookbook_name = nil)
+        set_or_return(:cookbook, cookbook_name, :kind_of => String)
+      end
+
+      def path(arg = @path)
+        set_or_return(:path, arg, :kind_of => String)
+      end
+
+      def hash(hash = nil)
+        set_or_return(:hash, hash, :kind_of => Hash)
+      end
+
+      def service_type(service_type = @service_type)
```

Comment:
When I wrote it this way I wanted to abstract the fact that launchd plists have two different types, and the only difference between the two are where they're placed. One is place in a location(/Library/LaunchDaemons/) and ran as root, the other is placed in a separate location(/Library/LaunchAgents/) and ran as user. And yes this is just overriding "path", because for some reason if someone wanted to create a launchd plist somewhere other than the normal locations I wanted to support that.